staging/src/k8s.io/apiserver/pkg/util/openstack/flavors.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2021 The Arktos Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import "fmt"
+
+var flavors = map[string]FlavorType{}
+
+// slim down version of the openstack flavor data structure
+type FlavorType struct {
+	Id          int
+	Name        string
+	MemoryMb    int
+	Vcpus       int
+	RootGb      int
+	EphemeralGb int
+}
+
+func initFlavorsCache() {
+	flavors = make(map[string]FlavorType)
+	flavors["m1.tiny"] = FlavorType{1, "m1.tiny", 512, 1, 0, 0}
+	flavors["m1.small"] = FlavorType{2, "m1.small", 2048, 1, 0, 0}
+	flavors["m1.medium"] = FlavorType{3, "m1.medium", 4096, 2, 0, 0}
+	flavors["m1.large"] = FlavorType{4, "m1.large", 8192, 4, 0, 0}
+	flavors["m1.xlarge"] = FlavorType{5, "m1.xlarge", 16384, 8, 0, 0}
+}
+
+func GetFalvor(name string) (*FlavorType, error) {
+	if flavor, found := flavors[name]; found {
+		return &flavor, nil
+	}
+
+	return nil, fmt.Errorf("flavor %s, not found", name)
+}
+
+func ListFalvors() []*FlavorType {
```

Comment:
This function returns array of pointers. No need to construct array every time.

Reply:
This is intentional to rebuild when the function is called. for couple of reasons:  flavors, images will be added in future; and also the map is used for the structure for quick GET() support hence the conventing to the slices from map values.

Reply:
In this code, all flavors are added at the init cache stage. There is no add function to support dynamic adding/delete flavors. I don't see why the list cannot be initialized and reused.

Reply:
fixed and add comment for the READ only support for now.